File: src/LibraryPlusPlus/Bootstrap.cpp
#include "Bootstrap.h"

DllExport void __stdcall Run(const wchar_t* args)
{
	const DWORD pid = GetCurrentProcessId();
	wchar_t buf[64];
	wsprintf(buf, L"Current Process: %d Args: %ls", pid, args);

	MessageBox(nullptr, buf, L"LibraryPlusPlus", NULL);
}

File: src/LibrarySharp/Bootstrap.cs
using System.Diagnostics;
using System.Runtime.InteropServices;
using System.Windows.Forms;
using RGiesecke.DllExport;

namespace LibrarySharp
{
    public static class Bootstrap
    {
        [DllExport("Run", CallingConvention.StdCall)]
        public static void Run([MarshalAs(UnmanagedType.LPWStr)] string args)
        {
            var process = Process.GetCurrentProcess();
            MessageBox.Show($"Current Process: {process.Id} Args: {args}", "LibrarySharp");
        }
    }
}
